native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js

```diff
@@ -16,185 +16,140 @@
-    } else {
-      iconSource = exclaimationLgIcon;
-    }
-
-
-    if (valid) {
-      style.push(styles.validOuter);
-      iconStyle.push(styles.validIcon);
-    } else {
-      style.push(styles.errorOuter);
-      iconStyle.push(styles.invalidIcon);
-    }
+
+  onBlur() {
+    this.setState(() => ({ isFocused: false }), this.props.onBlur);
   }
 
-  if (innerStyle) { style.push(innerStyle); }
-
-  return (
-    <View
-      style={style}
-    >
-      <TextInput
-        underlineColorAndroid="transparent"
-        placeholderTextColor={colorGray300}
-        style={textStyle}
-        editable={!disabled}
-        value={value}
-        ref={inputRef}
-        {...rest}
-      />
-      {iconSource &&
-        <Image
-          style={iconStyle}
-          source={iconSource}
-        />
-      }
-    </View>
-  );
-};
+  getColorAnimatedValue() {
+    return this.state.isFocused ? 1 : 0;
+  }
+
+  getLabelPositionAnimatedValue() {
+    return (this.props.value || this.state.isFocused) ? 0 : 1;
+  }
+
+  render() {
+    const { isFocused } = this.state;
+    const {
+      inputRef,
+      validationMessage,
+      editable,
+      label,
+      value,
+      style: userStyle,
+      valid,
+      onFocus,
+      onBlur,
+      ...rest
+    } = this.props;
+
+    const validityIcon = valid ? <ValidIcon /> : (valid === false && <InvalidIcon />);
+
+    const animatedLabelStyle = getLabelStyle(
+      this.animatedValues.color,
+      this.animatedValues.labelPosition,
+      { value, valid, editable },
+    );
+
+    const animatedInputStyle = getInputContainerStyle(this.animatedValues.color, valid);
+
+    return (
+      <View style={[styles.container, userStyle]}>
+        <Animated.Text style={animatedLabelStyle}>{label}</Animated.Text>
+        <Animated.View style={animatedInputStyle}>
+          <TextInput
+            editable={editable}
+            value={value || ''}
+            style={styles.input}
+            onFocus={this.onFocus}
+            onBlur={this.onBlur}
+            ref={inputRef}
+            underlineColorAndroid="transparent"
+            {...rest}
+          />
+          { !isFocused && validityIcon }
+        </Animated.View>
+        { valid === false && validationMessage && (
+          <BpkText textStyle="xs" style={styles.validationMessage}>
+            {validationMessage}
+          </BpkText>
+        )}
+      </View>
+    );
+  }
+}
 
 BpkTextInput.propTypes = {
-  disabled: PropTypes.bool,
-  small: PropTypes.bool,
-  style: TextInput.propTypes.style,
-  valid: PropTypes.bool,
+  label: PropTypes.string.isRequired,
   value: PropTypes.string.isRequired,
+  clearButtonMode: TextInput.propTypes.clearButtonMode,
   inputRef: PropTypes.func,
+  validationMessage: PropTypes.string,
+  valid: PropTypes.oneOf(true, false, null),
+  editable: PropTypes.bool,
+  style: ViewPropTypes.style,
+  onFocus: PropTypes.func,
+  onBlur: PropTypes.func,
+  onChangeText: PropTypes.func,
```

Comment:
Why is this documented? We aren't handling this directly as it's part of the spread

Reply:
Actually not sure - have removed.